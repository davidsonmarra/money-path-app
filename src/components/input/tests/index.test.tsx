import React from 'react';
import {render, screen} from '@testing-library/react-native';
import Input, {Props as InputProps} from 'src/components/input';

const renderComponent = ({...rest}: InputProps) => render(<Input {...rest} />);

describe('Input', () => {
  it('renders correctly', () => {
    const tree = renderComponent({}).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders placeholder correctly', () => {
    renderComponent({placeholder: 'placeholder'});
    expect(screen.getByPlaceholderText('placeholder')).toBeTruthy();
  });

  it('renders value correctly', () => {
    const input = renderComponent({value: 'value'}).root;
    expect(input.props.value).toBe('value');
  });

  it('renders disabled correctly', () => {
    const input = renderComponent({disabled: true}).root;
    expect(input.props.editable).toBeFalsy();
    expect(input.props.selectTextOnFocus).toBeFalsy();
  });

  it('should update input value', () => {
    let mockValue = 'new value';
    const mockOnChangeText = (value: string) => (mockValue = value);
    const input = renderComponent({
      value: mockValue,
      onChangeText: mockOnChangeText,
    }).root;
    input.props.onChangeText('new value');
    expect(input.props.value).toBe('new value');
  });
});
