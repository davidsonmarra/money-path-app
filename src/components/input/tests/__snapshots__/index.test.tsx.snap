// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#232E38",
      "borderColor": "#FFFFFF33",
      "borderRadius": 8,
      "borderWidth": 1,
      "flexDirection": "row",
      "paddingHorizontal": 16,
      "width": "100%",
    }
  }
>
  <TextInput
    editable={true}
    onBlur={[Function]}
    onFocus={[Function]}
    placeholderTextColor="#FFFFFF65"
    selectTextOnFocus={true}
    style={
      {
        "color": "#D9D9D9",
        "paddingVertical": 12,
        "width": "100%",
      }
    }
  />
</View>
`;
